### PR TITLE
Throw an exception if trying to do an ID server request with no ID server

### PR DESCRIPTION
**Pull request by dbkr**
Fixes https://github.com/vector-im/riot-web/issues/10540

### CODE REVIEW

File: src/http-api.js [hunk 1]
```diff
@@ -389,6 +389,10 @@ module.exports.MatrixHttpApi.prototype = {
         prefix,
         accessToken,
     ) {
+        if (!this.opts.idBaseUrl) {
+            throw new Error("No ID Server Base URL set");
```
Comment (jryans) [inline src/http-api.js]: Maybe let's be slightly more formal and say "identity server"?

```suggestion
            throw new Error("No Identity Server base URL set");
```

I also noticed this much more verbose error style:

https://github.com/matrix-org/matrix-js-sdk/blob/17724fc8d35f75d1a7a751b62da6db98798ef548/src/client.js#L2438-L2442

I guess it's just a historic thing we used to do for SDK errors a while ago?

File: src/http-api.js [hunk 2]
```diff
@@ -389,6 +389,10 @@ module.exports.MatrixHttpApi.prototype = {
         prefix,
         accessToken,
     ) {
+        if (!this.opts.idBaseUrl) {
```
Comment (jryans) [inline src/http-api.js]: We should also guard:

* `MatrixBaseApis.prototype.registerWithIdentityServer`

Also not quite sure what this one will do for empty string:

* `MatrixClient.prototype._requestTokenFromEndpoint`

**Comment by jryans** [comment]
It's probably good enough to upgrade the PR saying it fixes this issue. It's all we're likely to do about it at the moment I think...

File: src/http-api.js (hunk 1 above)
Comment (dbkr) [reply src/http-api.js]: Mmm, that looks like it's basically pretending to be a response from the server when it isn't, so not sure about copying that one.

File: src/http-api.js (hunk 2 above)
Comment (dbkr) [reply src/http-api.js]: url.parse will parse the empty string to null everything so the second one if fine as-is

**Pull request merged**